redwood-treehouse/src/hostMain/kotlin/app/cash/redwood/treehouse/EventListener.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.redwood.treehouse
+
+import app.cash.zipline.Call
+import app.cash.zipline.CallResult
+import app.cash.zipline.ZiplineService
+
+public abstract class EventListener {
+  public open fun appCreated(
+    app: TreehouseApp<*>,
+  ) {
+  }
+
+  public open fun appCanceled(
+    app: TreehouseApp<*>,
+  ) {
+  }
+
+  public open fun codeLoadStart(
+    app: TreehouseApp<*>,
+    manifestUrl: String?,
+  ): Any? = Unit
```

Comment:
Can we replace this with a strong type / alias like a `typealias TreehouseCodeLoadValue = Any?` just so it's more obvious what this does. Not sure what to name it

Reply:
What would we expect to return from this method? Shouldn't it be `{}` like the others?

Reply:
Was going to make the same comment. `null` seems best.

Reply:
Good call y’all. Changed to null.